kite-data/kite-data-crunch/src/main/java/org/kitesdk/data/crunch/CrunchDatasets.java

```diff
@@ -271,11 +303,14 @@ public void initialize() {
               .getProvidedValues();
         }
       }
+      count = 0;
     }
 
     @Override
-    public AvroStorageKey map(E entity) {
-      return key.reuseFor(entity, provided, accessor);
+    public Pair<GenericData.Record, Integer> map(E entity) {
```

Comment:
I didn't see that this uses Crunch's Pair class, rather than Kite's. That's why I was surprised it worked at first. :)